L1Trigger/L1TMuonEndCap/plugins/L1TMuonEndCapTrackProducer.cc

```diff
@@ -381,18 +381,26 @@ void L1TMuonEndCapTrackProducer::produce(edm::Event& ev,
   // Cancel out duplicate tracks
   for (unsigned int i1=0; i1 < AllTracks_PreCancel.size(); i1++) {
     bool dup = false;
-    int ebx = 20, sindex = -1;
+    int ebx = 20, sebx = 20, sindex = -1;
     for (std::vector<ConvertedHit>::iterator A = AllTracks_PreCancel[i1].AHits.begin(); A != AllTracks_PreCancel[i1].AHits.end(); A++) {
-      if (A->TP().getCSCData().bx < ebx) ebx = A->TP().getCSCData().bx;
+      if (A->TP().getCSCData().bx < ebx) {
+      	sebx = ebx;
+      	ebx = A->TP().getCSCData().bx;
+      }
+      else if (A->TP().getCSCData().bx < sebx) sebx = A->TP().getCSCData().bx;
       sindex = A->SectorIndex();
```

Comment:
Why is sindex assigned inside the for loop if it is only used after the loop with the value of the last hit?